webapp/views.py

```diff
@@ -445,29 +446,54 @@ def get_notices(**kwargs):
                 Notice.id.ilike(f"%{details}%"),
                 Notice.details.ilike(f"%{details}%"),
                 Notice.title.ilike(f"%{details}%"),
-                Notice.cves.any(CVE.id.ilike(f"%{details}%")),
             )
         )
 
     sort = asc if order_by == "oldest" else desc
 
-    notices = (
-        notices_query.options(
-            selectinload(Notice.cves).options(
-                selectinload(CVE.statuses),
-                selectinload(CVE.notices).options(
-                    load_only(
-                        Notice.id, Notice.is_hidden, Notice.release_packages
-                    )
-                ),
+    if cves:
+        # Get cves by numerical id
+        numerical_cve_ids = [
+            convert_cve_id_to_numerical_id(cve) for cve in cves
+        ]
+        matched_cves = (
+            db.session.query(CVE)
+            .filter(CVE.numerical_id.in_(numerical_cve_ids))
+            .all()
+        )
+        # Get notices_ids from cves
+        notice_ids = []
+        for cve in matched_cves:
+            notice_ids += [notice.id for notice in cve.notices]
+
+        notices = (
+            notices_query.filter(Notice.id.in_(notice_ids))
+            .order_by(sort(Notice.published), sort(Notice.id))
```

Comment:
This should fix the `total_results` field not returning the correct count:
```suggestion
        notices_query = notices_query.filter(Notice.id.in_(notice_ids))
        
        notices = (
                        notices_query.order_by(sort(Notice.published), sort(Notice.id))
```

Reply:
Thanks and will do!